net/network.go

```diff
@@ -53,3 +103,77 @@ func (network *Network) GetPeerID() peer.ID {
 func (network *Network) GetBandwidthStats() metrics.Stats {
 	return network.Reporter.GetBandwidthTotals()
 }
+
+// Connect connects to peers at the given addresses
```

Comment:
should say that it stops trying once it hits its first error. probably it should return results like the original did, but set the success field to false on connection error *and then keep trying the others*. i think maybe follow up with this change?